src/libs/actions/OnyxUpdateManager.js

```diff
@@ -0,0 +1,78 @@
+    Onyx.connect({
+        key: ONYXKEYS.ONYX_UPDATES_FROM_SERVER,
+        callback: (val) => {
+            if (!val) {
+                return;
+            }
+
+            const {lastUpdateIDFromServer, previousUpdateIDFromServer, updateParams} = val;
+
+            // In cases where we received a previousUpdateID and it doesn't match our lastUpdateIDAppliedToClient
+            // we need to perform one of the 2 possible cases:
+            //
+            // 1. This is the first time we're receiving an lastUpdateID, so we need to do a final reconnectApp before
+            // fully migrating to the reliable updates mode;
+            // 2. This this client already has the reliable updates mode enabled, but it's missing some updates and it
+            // needs to fetch those.
+            //
+            // To to both of those, we need to pause the sequential queue. This is important so that the updates are
+            // applied in their correct and specific order. If this queue was not paused, then there would be a lot of
+            // onyx data being applied while we are fetching the missing updates and that would put them all out of order.
+            SequentialQueue.pause();
+            let promise;
+
+            // The flow below is setting the promise to a reconnect app to address flow (1) explained above.
+            if (!lastUpdateIDAppliedToClient) {
+                console.debug('[OnyxUpdateManager] Client has not gotten reliable updates before so reconnecting the app to start the process');
+                Log.info('Client has not gotten reliable updates before so reconnecting the app to start the process');
+
+                // Since this is a full reconnectApp, we'll not apply the updates we received - those will come in the reconnect app request.
+                promise = App.lastReconnectAppAfterActivatingReliableUpdates();
+            } else {
+                // The flow below is setting the promise to a getMissingOnyxUpdates to address flow (2) explained above.
+                console.debug(`[OnyxUpdateManager] Client is behind the server by ${previousUpdateIDFromServer - lastUpdateIDAppliedToClient} so fetching incremental updates`);
+                Log.info('Gap detected in update IDs from server so fetching incremental updates', true, {
+                    lastUpdateIDFromServer,
+                    previousUpdateIDFromServer,
+                    lastUpdateIDAppliedToClient,
+                });
+                promise = App.getMissingOnyxUpdates(lastUpdateIDAppliedToClient, lastUpdateIDFromServer).then(() => {
+                    OnyxUpdates.apply({...updateParams, lastUpdateID: lastUpdateIDFromServer}).finally(() => {
+                        console.debug('[OnyxUpdateManager] Done applying all updates');
+                    });
+                });
+            }
+
+            promise.finally(() => {
+                SequentialQueue.unpause();
+            });
```

Comment:
```suggestion
            promise.finally(SequentialQueue.unpause);
```

Reply:
Done, but in another way since I had to chain those